numpy/core/tests/test_multiarray.py

```diff
@@ -4811,17 +4810,23 @@ def test_invalid_axis(self): # gh-7528
 class TestIO:
     """Test tofile, fromfile, tobytes, and fromstring"""
 
-    def setup(self):
+    @pytest.fixture()
+    def x(self):
```

Comment:
I was confused as to why this fixture had a parameter, but I guess that's because it's defined within a class. I've never seen that pattern used before. Alternatively this fixture could be defined outside of the class which would get rid of the confusing (and AFAICT unnecessary) parameter. Anyways, it's fine as-is I just wanted to bring it up in case this was intentional and I'd learn something new about fixtures :)

Reply:
You are correct, and this fixture could be brought outside the `TestIO` class. However, because it is only used within `TestIO`, it made more sense to declare it under `TestIO`. If `x` were to be used in other tests outside `TestIO`, then it could be defined at the module-level, which is definitely one of the benefits of using fixtures: easy reusability.

To get rid of the `self` parameter, I guess we can write:
```
    @staticmethod
    @pytest.fixture()
    def x():
        ...
```

Reply:
No worries, I was just curious - it's fine as-is.